app/src/main/java/memphis/myapplication/data/Common.java

```diff
@@ -327,38 +325,38 @@ public static void encryptAndPublish(Intent resultData, Context context, boolean
 
                 // Generate symmetric key
                 final SecretKey secretKey;
-                PublishedContent publishedContent = databaseViewModel.checkIfShared(path);
 
                 final byte[] iv = encrypter.generateIV();
 
                 // Encode sync data
-                SyncData syncData = new SyncData();
-                syncData.addLocation(location);
-                syncData.setIsFile(isFile);
+                Metadata metadata = new Metadata();
+                metadata.addLocation(location);
+                metadata.setIsFile(isFile);
 
                 final boolean feed = (recipients == null);
                 if (feed) {
                     secretKey = sharedPrefsManager.getKey();
                     Timber.d("For feed");
-                    syncData.setFeed(true);
+                    metadata.setFeed(true);
                 } else {
-                    if (publishedContent != null) {
-                        secretKey = publishedContent.getKey();
-                    }
-                    else
-                        secretKey = encrypter.generateKey();
-                    syncData.setFeed(false);
+                    secretKey = encrypter.generateKey();
+
+                    metadata.setFeed(false);
                     Timber.d("For friends");
-                    for (String friend : recipients) {
-                        Blob friendKey = databaseViewModel.getFriend(friend).getCert().getPublicKey();
-                        byte[] encryptedKey = RsaAlgorithm.encrypt
-                                (friendKey, new Blob(secretKey.getEncoded()), new EncryptParams(EncryptAlgorithmType.RsaOaep)).getImmutableArray();
-                        syncData.addFriendKey(friend, encryptedKey);
-                    }
+                    BloomFilter bloomFilter = new BloomFilter(recipients.size(), 0.001);
+                    for (String friend : recipients)
+                        bloomFilter.insert(friend);
+
+                    Name bloomName = new Name(bloomFilter.appendToName(new Name()));
+                    Timber.d("Size: " + bloomName.get(0).toNumber() + " Prob: " + bloomName.get(1).toNumber());
+                    metadata.setBloomFilter(bloomName);
+
                 }
-                filename = filename + "/" + Common.getKeyDigest(secretKey);
+                String keyDigest = Common.getKeyDigest(secretKey);
+                PublishedContent publishedContent = databaseViewModel.checkIfShared(keyDigest);
+                filename = filename + "/" + keyDigest;
```

Comment:
Is there some inbuilt class that handles creating file names?

Reply:
No. But there probably should be. A filename is <user-namespace>/file/<path>/<key-digest>.